p2pool/bitcoin/p2p.py

```diff
@@ -146,6 +146,16 @@ def handle_pong(self, nonce):
     ])
     def handle_alert(self, message, signature):
         pass # print 'ALERT:', (message, signature)
+
+    message_reject = pack.ComposedType([
+        ('message', pack.VarStrType()),
+        ('ccode', pack.IntType(8)),
+        ('reason', pack.VarStrType()),
+        ('data', pack.IntType(256)),
+    ])
+    def handle_reject(self, message, ccode, reason, data):
+        if p2pool.DEBUG:
+            print >>sys.stderr, 'Received reject message (%s): %s' % (message, reason)
```

Comment:
I've only used it when debugging the witness commitment calculation. You may be right though.